assets/src/block-editor/helpers/index.js

```diff
@@ -164,6 +164,66 @@ export const addAMPAttributes = ( settings, name ) => {
 	return settings;
 };
 
+/**
+ * Remove AMP deprecated props from any blocks which may contain them.
+ *
+ * @param {Object} settings Block settings.
+ *
+ * @return {Object} Modified block settings.
+ */
+export const removeDeprecatedAmpProps = ( settings ) => {
+	const mappedAttributes = {
+		ampCarousel: 'data-amp-carousel',
+		ampLightbox: 'data-amp-lightbox',
+		ampNoLoading: 'data-amp-noloading',
+		ampLayout: 'data-amp-layout',
+	};
+	const maybeBlockHasDeprecatedProp = Object.keys( mappedAttributes )
+		.some( ( attribute ) => Object.keys( settings.attributes ).includes( attribute ) );
+
+	if ( maybeBlockHasDeprecatedProp ) {
+		// At minimum, the `attributes`, and `save` props are required.
+		const deprecatedConfig = {
+			attributes: settings.attributes,
+			save( blockData ) {
+				const deprecatedProps = {};
+
+				// Render the block.
+				let element = settings.save( blockData );
+				const { attributes: blockAttributes } = blockData;
+
+				// Collect all deprecated props that would be set on the saved HTML block in the post.
+				Object.keys( mappedAttributes ).forEach( ( attribute ) => {
+					const deprecatedPropSet = Object.keys( blockAttributes ).includes( attribute ) && blockAttributes[ attribute ];
+
+					if ( deprecatedPropSet ) {
```

Comment:
Minor thing, but when I first read `deprecatedPropSet` I was thinking this was an instance of an actual `Set` object in JS. Perhaps either change is to `hasDeprecatedProp` or just do:

```suggestion
					if ( Object.keys( blockAttributes ).includes( attribute ) && blockAttributes[ attribute ] ) {
```

Reply:
Oh, but actually this can be simplified further to just:

```suggestion
					if ( attribute in blockAttributes ) {
```